x/twap/types/keys.go

```diff
@@ -92,3 +98,30 @@ func ParseTwapFromBz(bz []byte) (twap TwapRecord, err error) {
 	}
 	return twap, err
 }
+
+// ParseTwapHistoricalPoolIndexedRecordFromBz parses through an existing Twap Record by key and returns Twap Record.
+func ParseTwapHistoricalPoolIndexedRecordFromBz(key []byte, value []byte) (TwapRecord, error) {
```

Comment:
Why is key parsing needed here?

The record written in state must already have pool id set correctly.

I suspect we don't need this function and can get by with `ParseTwapFromBz`